src/mlpack/core/tree/hrectbound.hpp

```diff
@@ -229,13 +226,13 @@ class HRectBound
    * Serialize the bound object.
    */
   template<typename Archive>
-  void serialize(Archive& ar, const unsigned int version);
+  void serialize(Archive& ar);
 
  private:
   //! The dimensionality of the bound.
   size_t dim;
   //! The bounds for each dimension.
-  math::RangeType<ElemType>* bounds;
+  std::vector<math::RangeType<ElemType>> bounds;
```

Comment:
@rcurtin I have tried to use `cereal::make_array` but there was an error telling that cereal can not serialize raw pointers.
Also, I am not sure if the error was from `hrectbound` or from another class. 
I thought about the change since it is rare to see this raw pointer array these days in mlpack. Even, we can use `std::array` instead of `std::vector` if it is very slow.

Reply:
I'd prefer to avoid changing this unless you also want to benchmark the performance and show that the performances are comparable.  Many years ago quite some time went into the implementation of this class, since it is used very heavily throughout tree traversals, and at that time it was found that the fastest implementation could be done with raw pointers.

I thought that you had serialized raw pointer arrays elsewhere too?